esp-hal/src/interrupt/riscv.rs

```diff
@@ -908,4 +716,224 @@ mod plic {
 
         prev_interrupt_priority
     }
+
+    #[cfg(feature = "rt")]
+    mod rt {
+        use super::*;
+        /// Get interrupt priority - called by assembly code
+        #[unsafe(no_mangle)]
+        pub(super) unsafe extern "C" fn priority(cpu_interrupt: CpuInterrupt) -> Priority {
+            super::priority(cpu_interrupt)
+        }
+
+        #[unsafe(no_mangle)]
+        #[unsafe(link_section = ".trap")]
+        pub(super) unsafe extern "C" fn _handle_priority() -> u32 {
+            let interrupt_id: usize = riscv::register::mcause::read().code(); // MSB is whether its exception or interrupt.
+            let interrupt_priority = PLIC_MX::regs()
+                .mxint_pri(interrupt_id)
+                .read()
+                .cpu_mxint_pri()
+                .bits();
+
+            let prev_interrupt_priority = PLIC_MX::regs()
+                .mxint_thresh()
+                .read()
+                .cpu_mxint_thresh()
+                .bits();
+            if interrupt_priority < 15 {
+                // leave interrupts disabled if interrupt is of max priority.
+                PLIC_MX::regs()
+                    .mxint_thresh()
+                    .write(|w| unsafe { w.cpu_mxint_thresh().bits(interrupt_priority + 1) });
+
+                unsafe {
+                    riscv::interrupt::enable();
+                }
+            }
+            prev_interrupt_priority as u32
+        }
+
+        #[unsafe(no_mangle)]
+        #[unsafe(link_section = ".trap")]
+        pub(super) unsafe extern "C" fn _restore_priority(stored_prio: u32) {
+            riscv::interrupt::disable();
+            PLIC_MX::regs()
+                .mxint_thresh()
+                .write(|w| unsafe { w.cpu_mxint_thresh().bits(stored_prio as u8) });
+        }
+    }
+}
+
+#[cfg(feature = "rt")]
+mod rt {
+    use esp_riscv_rt::TrapFrame;
+
+    use super::*;
+
+    /// # Safety
+    ///
+    /// This function is called from an assembly trap handler.
+    #[doc(hidden)]
+    #[unsafe(link_section = ".trap.rust")]
+    #[unsafe(export_name = "_start_trap_rust_hal")]
+    unsafe extern "C" fn start_trap_rust_hal(trap_frame: *mut TrapFrame) {
+        assert!(
+            mcause::read().is_exception(),
+            "Arrived into _start_trap_rust_hal but mcause is not an exception!"
```

Comment:
```suggestion
            "Arrived into start_trap_rust_hal but mcause is not an exception!"
```

Reply:
This gets exported as `_start_trap_rust_hal`